lib/internal/http2/compat.js

```diff
@@ -659,24 +653,31 @@ class Http2ServerResponse extends Stream {
       encoding = 'utf8';
     }
 
+    if (state.closed) {
+      if (typeof cb === 'function')
```

Comment:
@mcollina @lpinca related https://github.com/nodejs/node/issues/28667

Reply:
It could be excluded and moved to another PR. Would you like me to remove it?

Reply:
@ronag .. yes, it would be best if you could separate out the unrelated changes